src/components/challenge-rewards-drawer/ChallengeRewardsDrawerProvider.tsx

```diff
@@ -202,6 +202,24 @@ export const ChallengeRewardsDrawerProvider = () => {
   const config = challengesConfig[modalType]
   const hasChallengeCompleted =
     challenge?.state === 'completed' || challenge?.state === 'disbursed'
+
+  // We could just depend on undisbursedAmount here
+  // But DN may have not indexed the challenge so check for client-side completion too
+  // Note that we can't handle aggregate challenges optimistically
+  let audioToClaim = 0
```

Comment:
Might be nice to pull this out to a function that accepts a challenge since we have this same logic in desktop

Reply:
Or just put this in OptimisticUserChallenge? Or am I overusing that too much? 😅